Build warnings from CI:

##  build (3.10, macos-15, >=2.1)

```
2026-03-15T12:53:54.1064430Z   apytypes 0.5.0.dev106+gd8d0936e0
2026-03-15T12:53:54.1065820Z 
2026-03-15T12:53:54.1067750Z     Subprojects
2026-03-15T12:53:54.1069430Z       bshoshany-thread-pool: YES
2026-03-15T12:53:54.1070970Z       fmt                  : YES
2026-03-15T12:53:54.1072550Z       highway              : YES
2026-03-15T12:53:54.1074040Z       nanobind             : YES
2026-03-15T12:53:54.1075730Z       robin-map            : YES (from nanobind)
2026-03-15T12:53:54.1077170Z 
2026-03-15T12:53:54.1079310Z     User defined options
2026-03-15T12:53:54.1081560Z       Native files         : /Users/runner/work/apytypes/apytypes/.mesonpy-5l5do7s_/meson-python-native-file.ini
2026-03-15T12:53:54.1083110Z       b_ndebug             : if-release
2026-03-15T12:53:54.1084680Z       b_vscrt              : md
2026-03-15T12:53:54.1086290Z       buildtype            : release
2026-03-15T12:53:54.1087700Z 
2026-03-15T12:53:54.1089730Z   Found ninja-1.13.2 at /opt/homebrew/bin/ninja
2026-03-15T12:53:54.1587020Z   + /opt/homebrew/bin/ninja
2026-03-15T12:53:55.4869470Z   [1/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_perf_counters.cc.o
2026-03-15T12:53:55.5486530Z   [2/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_abort.cc.o
2026-03-15T12:53:55.5990070Z   [3/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_profiler.cc.o
2026-03-15T12:53:55.6413260Z   [4/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_aligned_allocator.cc.o
2026-03-15T12:53:56.4013130Z   [5/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_nanobenchmark.cc.o
2026-03-15T12:53:56.4315430Z   [6/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_print.cc.o
2026-03-15T12:53:56.8054480Z   [7/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_timer.cc.o
2026-03-15T12:53:57.5202160Z   [8/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_targets.cc.o
2026-03-15T12:53:58.5761490Z   [9/35] Compiling C++ object subprojects/highway-1.3.0/libhwy.a.p/hwy_per_target.cc.o
2026-03-15T12:53:59.8281080Z   [10/35] Linking static target subprojects/highway-1.3.0/libhwy.a
2026-03-15T12:54:01.5449770Z   [11/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfixed_wrapper.cc.o
2026-03-15T12:54:03.4387160Z   [12/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfixed.cc.o
2026-03-15T12:54:05.8201610Z   [13/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfloatarray_wrapper.cc.o
2026-03-15T12:54:08.4770080Z   [14/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfloatarray.cc.o
2026-03-15T12:54:08.4871340Z   In file included from ../src/apycfloatarray.cc:3:
2026-03-15T12:54:08.4973990Z   In file included from ../src/apycfloat_util.h:5:
2026-03-15T12:54:08.5081180Z   ../src/apyfloat_util.h:2276:36: warning: definition of implicit copy constructor for '_FloatingPointMultiplierShort' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
2026-03-15T12:54:08.5088700Z    2276 |     _FloatingPointMultiplierShort& operator=(const _FloatingPointMultiplierShort&)
2026-03-15T12:54:08.5089300Z         |                                    ^
2026-03-15T12:54:08.5090380Z   ../src/apycfloat_util.h:424:8: note: in implicit copy constructor for '_FloatingPointMultiplierShort' first required here
2026-03-15T12:54:08.5291850Z     424 | struct ComplexFloatingPointInnerProduct {
2026-03-15T12:54:08.5293040Z         |        ^
2026-03-15T12:54:08.5301750Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/allocator.h:165:24: note: in implicit copy constructor for 'ComplexFloatingPointInnerProduct' first required here
2026-03-15T12:54:08.5413470Z     165 |     ::new ((void*)__p) _Up(std::forward<_Args>(__args)...);
2026-03-15T12:54:08.5519550Z         |                        ^
2026-03-15T12:54:08.5622960Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/allocator_traits.h:319:9: note: in instantiation of function template specialization 'std::allocator<ComplexFloatingPointInnerProduct>::construct<ComplexFloatingPointInnerProduct, const ComplexFloatingPointInnerProduct &>' requested here
2026-03-15T12:54:08.5728000Z     319 |     __a.construct(__p, std::forward<_Args>(__args)...);
2026-03-15T12:54:08.5830020Z         |         ^
2026-03-15T12:54:08.5935150Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:1149:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<ComplexFloatingPointInnerProduct>>::construct<ComplexFloatingPointInnerProduct, const ComplexFloatingPointInnerProduct &, 0>' requested here
2026-03-15T12:54:08.6039850Z    1149 |     __alloc_traits::construct(this->__alloc(), std::__to_address(__pos), __x);
2026-03-15T12:54:08.6162220Z         |                     ^
2026-03-15T12:54:08.6164750Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:466:7: note: in instantiation of member function 'std::vector<ComplexFloatingPointInnerProduct>::__construct_at_end' requested here
2026-03-15T12:54:08.6268450Z     466 |       __construct_at_end(__n, __x);
2026-03-15T12:54:08.6370750Z         |       ^
2026-03-15T12:54:08.6472400Z   ../src/apycfloatarray.cc:1430:55: note: in instantiation of member function 'std::vector<ComplexFloatingPointInnerProduct>::vector' requested here
2026-03-15T12:54:08.6574120Z    1430 |         std::vector<ComplexFloatingPointInnerProduct> cache_inner_prod(
2026-03-15T12:54:08.6676390Z         |                                                       ^
2026-03-15T12:54:08.6778930Z   1 warning generated.
2026-03-15T12:54:10.7221830Z   [15/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfloat_wrapper.cc.o
2026-03-15T12:54:11.4602450Z   [16/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfloat.cc.o
2026-03-15T12:54:11.7845650Z   [17/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfixedarray_wrapper.cc.o
2026-03-15T12:54:13.7465110Z   [18/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apycfixedarray.cc.o
2026-03-15T12:54:15.0025680Z   [19/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfixed_wrapper.cc.o
2026-03-15T12:54:16.7426820Z   [20/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfixedarray_iterator.cc.o
2026-03-15T12:54:17.3771300Z   [21/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfixed.cc.o
2026-03-15T12:54:20.3915330Z   [22/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfloat_wrapper.cc.o
2026-03-15T12:54:21.4406870Z   [23/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfloat.cc.o
2026-03-15T12:54:23.0186780Z   [24/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfixedarray_wrapper.cc.o
2026-03-15T12:54:23.6844300Z   [25/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfixedarray.cc.o
2026-03-15T12:54:24.9710680Z   [26/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfloatarray_iterator.cc.o
2026-03-15T12:54:25.3331000Z   [27/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apytypes_context_wrapper.cc.o
2026-03-15T12:54:25.7219000Z   [28/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apytypes_mp.cc.o
2026-03-15T12:54:27.8125310Z   [29/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apytypes_common.cc.o
2026-03-15T12:54:28.7467150Z   [30/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apytypes_wrapper.cc.o
2026-03-15T12:54:29.4327770Z   [31/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfloatarray.cc.o
2026-03-15T12:54:29.4429810Z   In file included from ../src/apyfloatarray.cc:4:
2026-03-15T12:54:29.4533120Z   ../src/apyfloat_util.h:2276:36: warning: definition of implicit copy constructor for '_FloatingPointMultiplierShort' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
2026-03-15T12:54:29.4636020Z    2276 |     _FloatingPointMultiplierShort& operator=(const _FloatingPointMultiplierShort&)
2026-03-15T12:54:29.4687250Z         |                                    ^
2026-03-15T12:54:29.4697490Z   ../src/apyfloat_util.h:2819:7: note: in implicit copy constructor for '_FloatingPointMultiplierShort' first required here
2026-03-15T12:54:29.4698170Z    2819 | class FloatingPointInnerProduct {
2026-03-15T12:54:29.4698500Z         |       ^
2026-03-15T12:54:29.4700050Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/allocator.h:165:24: note: in implicit copy constructor for 'FloatingPointInnerProduct' first required here
2026-03-15T12:54:29.4701790Z     165 |     ::new ((void*)__p) _Up(std::forward<_Args>(__args)...);
2026-03-15T12:54:29.4702250Z         |                        ^
2026-03-15T12:54:29.4703520Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/allocator_traits.h:319:9: note: in instantiation of function template specialization 'std::allocator<FloatingPointInnerProduct>::construct<FloatingPointInnerProduct, const FloatingPointInnerProduct &>' requested here
2026-03-15T12:54:29.4704720Z     319 |     __a.construct(__p, std::forward<_Args>(__args)...);
2026-03-15T12:54:29.4704960Z         |         ^
2026-03-15T12:54:29.4706040Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:1149:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<FloatingPointInnerProduct>>::construct<FloatingPointInnerProduct, const FloatingPointInnerProduct &, 0>' requested here
2026-03-15T12:54:29.4707930Z    1149 |     __alloc_traits::construct(this->__alloc(), std::__to_address(__pos), __x);
2026-03-15T12:54:29.4708300Z         |                     ^
2026-03-15T12:54:29.4709120Z   /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:466:7: note: in instantiation of member function 'std::vector<FloatingPointInnerProduct>::__construct_at_end' requested here
2026-03-15T12:54:29.4710030Z     466 |       __construct_at_end(__n, __x);
2026-03-15T12:54:29.4710340Z         |       ^
2026-03-15T12:54:29.4711070Z   ../src/apyfloatarray.cc:1879:48: note: in instantiation of member function 'std::vector<FloatingPointInnerProduct>::vector' requested here
2026-03-15T12:54:29.4712160Z    1879 |         std::vector<FloatingPointInnerProduct> cache_inner_prod(n_threads, inner_prod);
2026-03-15T12:54:29.4712910Z         |                                                ^
2026-03-15T12:54:29.4713200Z   1 warning generated.
2026-03-15T12:54:29.9433820Z   [32/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/subprojects_nanobind-2.12.0_src_nb_combined.cpp.o
2026-03-15T12:54:30.1303690Z   [33/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apyfloatarray_wrapper.cc.o
2026-03-15T12:54:30.4605290Z   [34/35] Compiling C++ object _apytypes.cpython-310-darwin.so.p/src_apytypes_simd.cc.o
2026-03-15T12:54:41.9639260Z   [35/35] Linking target _apytypes.cpython-310-darwin.so
2026-03-15T12:54:41.9639800Z   ld: warning: -U option is redundant when using -undefined dynamic_lookup
2026-03-15T12:54:42.0972600Z   [1/9] /Users/runner/work/apytypes/apytypes/.mesonpy-5l5do7s_/_apytypes.cpython-310-darwin.so
2026-03-15T12:54:42.0976500Z   [2/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/__init__.py
2026-03-15T12:54:42.0978360Z   [3/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/_apytypes.pyi
2026-03-15T12:54:42.0980620Z   [4/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/_array_functions.py
2026-03-15T12:54:42.0999260Z   [5/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/typing.py
2026-03-15T12:54:42.0999950Z   [6/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/_utils.py
2026-03-15T12:54:42.1000540Z   [7/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/vhdl.py
2026-03-15T12:54:42.1001000Z   [8/9] /Users/runner/work/apytypes/apytypes/lib/apytypes/py.typed
2026-03-15T12:54:42.1001420Z   [9/9] /Users/runner/work/apytypes/apytypes/.mesonpy-5l5do7s_/_version.py
2026-03-15T12:54:42.1096350Z   Preparing metadata (pyproject.toml): finished with status 'done'
2026-03-15T12:54:42.1242530Z Requirement already satisfied: numpy>=1.25 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from apytypes==0.5.0.dev106+gd8d0936e0) (2.2.6)
2026-03-15T12:54:42.1943740Z Collecting pytest>=8.2 (from apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.1946060Z   Obtaining dependency information for pytest>=8.2 from https://files.pythonhosted.org/packages/3b/ab/b3226f0bd7cdcf710fbede2b3548584366da3b19b5021e74f5bde2a8fa3f/pytest-9.0.2-py3-none-any.whl.metadata
2026-03-15T12:54:42.2106840Z   Downloading pytest-9.0.2-py3-none-any.whl.metadata (7.6 kB)
2026-03-15T12:54:42.2274290Z Collecting exceptiongroup>=1 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.2275610Z   Obtaining dependency information for exceptiongroup>=1 from https://files.pythonhosted.org/packages/8a/0e/97c33bf5009bdbac74fd2beace167cab3f978feb69cc36f1ef79360d6c4e/exceptiongroup-1.3.1-py3-none-any.whl.metadata
2026-03-15T12:54:42.2309260Z   Downloading exceptiongroup-1.3.1-py3-none-any.whl.metadata (6.7 kB)
2026-03-15T12:54:42.2409840Z Collecting iniconfig>=1.0.1 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.2410960Z   Obtaining dependency information for iniconfig>=1.0.1 from https://files.pythonhosted.org/packages/cb/b1/3846dd7f199d53cb17f49cba7e651e9ce294d8497c8c150530ed11865bb8/iniconfig-2.3.0-py3-none-any.whl.metadata
2026-03-15T12:54:42.2442220Z   Downloading iniconfig-2.3.0-py3-none-any.whl.metadata (2.5 kB)
2026-03-15T12:54:42.2541440Z Collecting packaging>=22 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.2542490Z   Obtaining dependency information for packaging>=22 from https://files.pythonhosted.org/packages/b7/b9/c538f279a4e237a006a2c98387d081e9eb060d203d8ed34467cc0f0b9b53/packaging-26.0-py3-none-any.whl.metadata
2026-03-15T12:54:42.2562590Z   Using cached packaging-26.0-py3-none-any.whl.metadata (3.3 kB)
2026-03-15T12:54:42.2714530Z Collecting pluggy<2,>=1.5 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.2715480Z   Obtaining dependency information for pluggy<2,>=1.5 from https://files.pythonhosted.org/packages/54/20/4d324d65cc6d9205fabedc306948156824eb9f0ee1633355a8f7ec5c66bf/pluggy-1.6.0-py3-none-any.whl.metadata
2026-03-15T12:54:42.2749110Z   Downloading pluggy-1.6.0-py3-none-any.whl.metadata (4.8 kB)
2026-03-15T12:54:42.2885270Z Collecting pygments>=2.7.2 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.2886470Z   Obtaining dependency information for pygments>=2.7.2 from https://files.pythonhosted.org/packages/c7/21/705964c7812476f378728bdf590ca4b771ec72385c533964653c68e86bdc/pygments-2.19.2-py3-none-any.whl.metadata
2026-03-15T12:54:42.2915670Z   Downloading pygments-2.19.2-py3-none-any.whl.metadata (2.5 kB)
2026-03-15T12:54:42.3034590Z Collecting tomli>=1 (from pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.3035630Z   Obtaining dependency information for tomli>=1 from https://files.pythonhosted.org/packages/23/d1/136eb2cb77520a31e1f64cbae9d33ec6df0d78bdf4160398e86eec8a8754/tomli-2.4.0-py3-none-any.whl.metadata
2026-03-15T12:54:42.3051910Z   Using cached tomli-2.4.0-py3-none-any.whl.metadata (10 kB)
2026-03-15T12:54:42.3169310Z Collecting typing-extensions>=4.6.0 (from exceptiongroup>=1->pytest>=8.2->apytypes==0.5.0.dev106+gd8d0936e0)
2026-03-15T12:54:42.3170600Z   Obtaining dependency information for typing-extensions>=4.6.0 from https://files.pythonhosted.org/packages/18/67/36e9267722cc04a6b9f15c7f3441c2363321a3ea07da7ae0c0707beb2a9c/typing_extensions-4.15.0-py3-none-any.whl.metadata
2026-03-15T12:54:42.3214050Z   Downloading typing_extensions-4.15.0-py3-none-any.whl.metadata (3.3 kB)
2026-03-15T12:54:42.3304100Z Downloading pytest-9.0.2-py3-none-any.whl (374 kB)
2026-03-15T12:54:42.3413660Z Downloading pluggy-1.6.0-py3-none-any.whl (20 kB)
2026-03-15T12:54:42.3485430Z Downloading exceptiongroup-1.3.1-py3-none-any.whl (16 kB)
2026-03-15T12:54:42.3559210Z Downloading iniconfig-2.3.0-py3-none-any.whl (7.5 kB)
2026-03-15T12:54:42.3612810Z Using cached packaging-26.0-py3-none-any.whl (74 kB)
2026-03-15T12:54:42.3660930Z Downloading pygments-2.19.2-py3-none-any.whl (1.2 MB)
2026-03-15T12:54:42.3825550Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 68.4 MB/s  0:00:00
2026-03-15T12:54:42.3847620Z Using cached tomli-2.4.0-py3-none-any.whl (14 kB)
2026-03-15T12:54:42.3890410Z Downloading typing_extensions-4.15.0-py3-none-any.whl (44 kB)
2026-03-15T12:54:42.3990770Z Building wheels for collected packages: apytypes
2026-03-15T12:54:42.3999730Z   Building wheel for apytypes (pyproject.toml): started
2026-03-15T12:54:42.4000500Z   Running command Building wheel for apytypes (pyproject.toml)
2026-03-15T12:54:42.4467160Z   Building wheel for apytypes (pyproject.toml): finished with status 'done'
2026-03-15T12:54:42.4478710Z   Created wheel for apytypes: filename=apytypes-0.5.0.dev106+gd8d0936e0-cp310-cp310-macosx_15_0_arm64.whl size=818118 sha256=2bc344907559c767d4ca133952618127c9b3cdd830c5086aec95c60535752ce2
2026-03-15T12:54:42.4479830Z   Stored in directory: /private/var/folders/t5/f77_gwnj6p95qxy9py3fckx00000gn/T/pip-ephem-wheel-cache-eyx3ba_r/wheels/b7/1c/cf/18251ebaf3ded190feb172ffdcfcf9b4745601ec42ed59caec
2026-03-15T12:54:42.4495890Z Successfully built apytypes
2026-03-15T12:54:42.4615910Z Installing collected packages: typing-extensions, tomli, pygments, pluggy, packaging, iniconfig, apytypes, exceptiongroup, pytest
2026-03-15T12:54:42.9524210Z   changing mode of /Library/Frameworks/Python.framework/Versions/3.10/bin/pygmentize to 755
2026-03-15T12:54:43.1961350Z   changing mode of /Library/Frameworks/Python.framework/Versions/3.10/bin/py.test to 755
2026-03-15T12:54:43.1966470Z   changing mode of /Library/Frameworks/Python.framework/Versions/3.10/bin/pytest to 755
2026-03-15T12:54:43.1988570Z 
2026-03-15T12:54:43.2002940Z Successfully installed apytypes-0.5.0.dev106+gd8d0936e0 exceptiongroup-1.3.1 iniconfig-2.3.0 packaging-26.0 pluggy-1.6.0 pygments-2.19.2 pytest-9.0.2 tomli-2.4.0 typing-extensions-4.15.0
```

--- a/src/apyfixed_util.h
+++ b/src/apyfixed_util.h
@@ -1225,13 +1225,13 @@ fixed_point_from_double_single_limb(double value, int frac_bits, unsigned shift_
         }
     }
 
+    apy_limb_signed_t sman = apy_limb_signed_t(man);
     if (sign_of_double(value)) {
-        man = -man;
+        sman = -sman;
     }
 
     // Two's complement overflow
-    man = apy_limb_t(apy_limb_signed_t(man << shift_amount) >> shift_amount);
-    return man;
+    return apy_limb_t((sman << shift_amount) >> shift_amount);
 }
 
 template <typename RANDOM_ACCESS_IT>

--- a/src/apyfloat_util.h
+++ b/src/apyfloat_util.h
@@ -1374,7 +1374,7 @@ template <
         my_aligned = (my != 0);
     } else {
         bool round_bit = ((my << (_MAN_T_SIZE_BITS - exp_delta)) != 0);
-        my_aligned = (my >> exp_delta) | round_bit;
+        my_aligned = (my >> exp_delta) | (unsigned int)round_bit;
     }
 
     // Perform addition / subtraction
@@ -1483,7 +1483,7 @@ template <
 
     const std::int64_t x_true_exp = true_exp(x_wide, x_spec.bias);
     const std::int64_t y_true_exp = true_exp(y_wide, y_spec.bias);
-    const unsigned exp_delta = x_true_exp - y_true_exp;
+    const std::uint64_t exp_delta = x_true_exp - y_true_exp;
 
     if (exp_delta == 0 && !same_sign && x_wide.man == y_wide.man) {
         man_t res_man = qntz == QuantizationMode::JAM ? 1 : 0;
@@ -2271,10 +2271,6 @@ public:
         BIAS_TERM = dst_spec.bias - std::int64_t(src1_spec.bias)
             - std::int64_t(src2_spec.bias);
     }
-
-    // Copy assign a new functor in the place of this one
-    _FloatingPointMultiplierShort& operator=(const _FloatingPointMultiplierShort&)
-        = default;
 
     void operator()(
         const APyFloatData* src1, const APyFloatData* src2, APyFloatData* dst

--- a/src/apytypes_mp.h
+++ b/src/apytypes_mp.h
@@ -9,6 +9,7 @@
 #include <cstddef>  // std::size_t
 #include <cstdint>  // std::int64_t, std::uint64_t, std::int32_t, std::uint32_t
 #include <iterator> // std::distance
+#include <vector>   // std::vector
 
 #include "apytypes_fwd.h"
 #include "apytypes_intrinsics.h"
@@ -26,8 +27,8 @@ template <class RANDOM_ACCESS_ITERATOR_INOUT, class RANDOM_ACCESS_ITERATOR_IN>
     RANDOM_ACCESS_ITERATOR_IN src_end
 )
 {
-    assert(dest_begin != dest_end);
-    assert(src_begin != src_end);
+    assert(dest_begin < dest_end);
+    assert(src_begin < src_end);
     assert(std::distance(dest_begin, dest_end) >= std::distance(src_begin, src_end));
 
     apy_limb_t carry = 0;
@@ -81,7 +82,7 @@ template <
     RANDOM_ACCESS_ITERATOR_IN2 src1
 )
 {
-    assert(dest_begin != dest_end);
+    assert(dest_begin < dest_end);
 
     // Specialized first iteration
     *dest_begin = *src0 + *src1;
@@ -111,18 +112,18 @@ template <
 }
 
 //! Add a single limb to a limb vector in place: dest += src
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_addition_single_limb(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it,
-    RANDOM_ACCESS_ITERATOR_IN cend_it,
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it,
+    RANDOM_ACCESS_ITERATOR_INOUT end_it,
     const apy_limb_t src
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
 
     /* src is initial "carry" */
     apy_limb_t carry = src;
-    for (auto it = cbegin_it; it != cend_it; ++it) {
+    for (auto it = begin_it; it != end_it; ++it) {
         *it += carry;
         carry = (*it < carry);
     }
@@ -131,15 +132,15 @@ template <class RANDOM_ACCESS_ITERATOR_IN>
 }
 
 //! Add a single one to a limb vector in place: dest++
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_add_one_lsb(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it, RANDOM_ACCESS_ITERATOR_IN cend_it
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it, RANDOM_ACCESS_ITERATOR_INOUT end_it
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
 
     apy_limb_t carry = 1;
-    for (auto it = cbegin_it; it != cend_it; ++it) {
+    for (auto it = begin_it; it != end_it; ++it) {
         *it += carry;
         carry = (*it < carry);
     }
@@ -156,7 +157,7 @@ template <class RANDOM_ACCESS_ITERATOR_INOUT, class RANDOM_ACCESS_ITERATOR_IN>
     RANDOM_ACCESS_ITERATOR_IN src_begin
 )
 {
-    assert(dest_begin != dest_end);
+    assert(dest_begin < dest_end);
 
     apy_limb_t carry = 0;
 
@@ -202,15 +203,15 @@ apy_inplace_addition_length_two(apy_limb_t* dest, const apy_limb_t* src)
 // Subtraction
 
 //! Negate a limb vector in place: dest = -dest
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_negate(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it, RANDOM_ACCESS_ITERATOR_IN cend_it
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it, RANDOM_ACCESS_ITERATOR_INOUT end_it
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
 
     apy_limb_t carry = 1;
-    for (auto it = cbegin_it; it != cend_it; ++it) {
+    for (auto it = begin_it; it != end_it; ++it) {
         *it = ~(*it) + carry;
         carry = (*it < carry);
     }
@@ -225,7 +226,7 @@ template <class RANDOM_ACCESS_ITERATOR_IN, class RANDOM_ACCESS_ITERATOR_OUT>
     RANDOM_ACCESS_ITERATOR_OUT dest_begin
 )
 {
-    assert(src_begin != src_end);
+    assert(src_begin < src_end);
 
     apy_limb_t carry = 1;
     auto dest_it = dest_begin;
@@ -240,19 +241,19 @@ template <class RANDOM_ACCESS_ITERATOR_IN, class RANDOM_ACCESS_ITERATOR_OUT>
 }
 
 //! Subtract a single limb from a limb vectors in place: dest -= src
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_subtraction_single_limb(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it,
-    RANDOM_ACCESS_ITERATOR_IN cend_it,
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it,
+    RANDOM_ACCESS_ITERATOR_INOUT end_it,
     const apy_limb_t src
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
 
     /* src is initial "carry" */
     apy_limb_t carry = src;
 
-    for (auto it = cbegin_it; it != cend_it; ++it) {
+    for (auto it = begin_it; it != end_it; ++it) {
         /* Determine carry to next limb */
         apy_limb_t carry_tmp = *it < carry;
         *it -= carry;
@@ -263,14 +264,14 @@ template <class RANDOM_ACCESS_ITERATOR_IN>
 }
 
 //! Subtract two limb vectors in place: dest -= src, where len(dest) == len(src)
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT, class RANDOM_ACCESS_ITERATOR_IN>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_subtraction_same_length(
-    RANDOM_ACCESS_ITERATOR_IN dest_begin,
-    RANDOM_ACCESS_ITERATOR_IN dest_end,
+    RANDOM_ACCESS_ITERATOR_INOUT dest_begin,
+    RANDOM_ACCESS_ITERATOR_INOUT dest_end,
     RANDOM_ACCESS_ITERATOR_IN src_begin
 )
 {
-    assert(dest_begin != dest_end);
+    assert(dest_begin < dest_end);
 
     apy_limb_t carry = 0;
     auto dest_it = dest_begin;
@@ -292,7 +293,7 @@ apy_inplace_reversed_subtraction_same_length(
     RANDOM_ACCESS_ITERATOR_IN src_begin
 )
 {
-    assert(dest_begin != dest_end);
+    assert(dest_begin < dest_end);
 
     // Specialized first iteration
     apy_limb_t carry = 1;
@@ -344,29 +345,29 @@ apy_inplace_reversed_subtraction_same_length(
 //! Left-shift limb vector in place
 apy_limb_t apy_inplace_left_shift(apy_limb_t*, const std::size_t, unsigned int);
 //! Left-shift limb vector in place
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 [[maybe_unused]] static APY_INLINE apy_limb_t apy_inplace_left_shift(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it,
-    RANDOM_ACCESS_ITERATOR_IN cend_it,
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it,
+    RANDOM_ACCESS_ITERATOR_INOUT end_it,
     const unsigned int shift_amount
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
     assert(shift_amount > 0);
     assert(shift_amount < APY_LIMB_SIZE_BITS);
 
     const unsigned int overlap = APY_LIMB_SIZE_BITS - shift_amount;
-    auto it = cend_it;
+    auto it = end_it;
     --it;
     apy_limb_t high_limb = *it;
     const apy_limb_t retval = high_limb >> overlap;
     apy_limb_t low_limb = (high_limb << shift_amount);
-    for (; it != cbegin_it; --it) {
+    for (; it != begin_it; --it) {
         high_limb = *(it - 1);
         *it = low_limb | (high_limb >> overlap);
         low_limb = (high_limb << shift_amount);
     }
-    *cbegin_it = low_limb;
+    *begin_it = low_limb;
 
     return retval;
 }
@@ -382,27 +383,27 @@ apy_limb_t apy_left_shift(
 apy_limb_t apy_inplace_right_shift(apy_limb_t*, const std::size_t, unsigned int);
 
 //! Right-shift limb vector in place
-template <class RANDOM_ACCESS_ITERATOR_IN>
+template <class RANDOM_ACCESS_ITERATOR_INOUT>
 apy_limb_t apy_inplace_right_shift(
-    RANDOM_ACCESS_ITERATOR_IN cbegin_it,
-    RANDOM_ACCESS_ITERATOR_IN cend_it,
+    RANDOM_ACCESS_ITERATOR_INOUT begin_it,
+    RANDOM_ACCESS_ITERATOR_INOUT end_it,
     unsigned int shift_amount
 )
 {
-    assert(cbegin_it != cend_it);
+    assert(begin_it < end_it);
     assert(shift_amount > 0);
     assert(shift_amount < APY_LIMB_SIZE_BITS);
 
     const unsigned int overlap = APY_LIMB_SIZE_BITS - shift_amount;
-    apy_limb_t low_limb = *cbegin_it;
+    apy_limb_t low_limb = *begin_it;
     const apy_limb_t retval = (low_limb << overlap);
     apy_limb_t high_limb = low_limb >> shift_amount;
-    for (auto it = cbegin_it + 1; it != cend_it; ++it) {
+    for (auto it = begin_it + 1; it != end_it; ++it) {
         low_limb = *it;
         *(it - 1) = high_limb | (low_limb << overlap);
         high_limb = low_limb >> shift_amount;
     }
-    *(cend_it - 1) = high_limb;
+    *(end_it - 1) = high_limb;
 
     return retval;
 }
@@ -458,7 +459,8 @@ void apy_unsigned_division(
 {
     auto denominator_limbs = std::distance(denominator_begin, denominator_end);
     auto numerator_limbs = std::distance(numerator_begin, numerator_end);
-    assert(denominator_begin != denominator_end);
+    assert(denominator_begin < denominator_end);
+    assert(numerator_begin < numerator_end);
     assert(numerator_limbs >= denominator_limbs);
 
     auto inv = APyDivInverse(denominator_begin.data(), denominator_limbs);
@@ -505,6 +507,8 @@ void apy_unsigned_division_preinverted(
 {
     auto denominator_limbs = std::distance(denominator_begin, denominator_end);
     auto numerator_limbs = std::distance(numerator_begin, numerator_end);
+    assert(numerator_begin < numerator_end);
+    assert(denominator_begin < denominator_end);
     assert(denominator_limbs > 0);
     assert(numerator_limbs >= denominator_limbs);
 

--- a/src/apytypes_util.h
+++ b/src/apytypes_util.h
@@ -12,7 +12,7 @@
 #include <fmt/format.h>
 
 #include <algorithm>        // std::find, std::unique, etc...
-#include <cstddef>          // std::size_t
+#include <cstddef>          // std::size_t, std::ptrdiff_t
 #include <cstdint>          // std::int64_t
 #include <functional>       // std::bit_not
 #include <initializer_list> // std::initializer_list
@@ -80,6 +80,8 @@ template <class RANDOM_ACCESS_ITERATOR>
 [[maybe_unused, nodiscard]] static APY_INLINE std::size_t
 significant_limbs(RANDOM_ACCESS_ITERATOR begin, RANDOM_ACCESS_ITERATOR end)
 {
+    assert(begin <= end);
+
     auto is_non_zero = [](auto n) { return n != 0; };
     auto back_non_zero_it = std::find_if(
         std::reverse_iterator(end), std::reverse_iterator(begin), is_non_zero
@@ -92,6 +94,8 @@ template <class RANDOM_ACCESS_ITERATOR>
 [[maybe_unused, nodiscard]] static APY_INLINE std::size_t
 limb_vector_leading_zeros(RANDOM_ACCESS_ITERATOR begin, RANDOM_ACCESS_ITERATOR end)
 {
+    assert(begin <= end);
+
     auto is_non_zero = [](auto n) { return n != 0; };
     auto rev_non_zero_it = std::find_if(
         std::reverse_iterator(end), std::reverse_iterator(begin), is_non_zero
@@ -106,11 +110,13 @@ limb_vector_leading_zeros(RANDOM_ACCESS_ITERATOR begin, RANDOM_ACCESS_ITERATOR e
     }
 }
 
-//! Return the number of leading zeros of a limb vector
+//! Return the number of trailing zeros of a limb vector
 template <class RANDOM_ACCESS_ITERATOR>
 [[maybe_unused, nodiscard]] static APY_INLINE std::size_t
 limb_vector_trailing_zeros(RANDOM_ACCESS_ITERATOR begin, RANDOM_ACCESS_ITERATOR end)
 {
+    assert(begin <= end);
+
     auto is_non_zero = [](auto n) { return n != 0; };
     auto non_zero_it = std::find_if(begin, end, is_non_zero);
     std::size_t zero_limbs = std::distance(begin, non_zero_it);
@@ -128,6 +134,8 @@ template <class RANDOM_ACCESS_ITERATOR>
 [[maybe_unused, nodiscard]] static APY_INLINE std::size_t
 limb_vector_leading_ones(RANDOM_ACCESS_ITERATOR begin, RANDOM_ACCESS_ITERATOR end)
 {
+    assert(begin <= end);
+
     auto is_not_all_ones = [](auto n) { return n != apy_limb_t(-1); };
     auto rev_not_all_ones_it = std::find_if(
         std::reverse_iterator(end), std::reverse_iterator(begin), is_not_all_ones
@@ -494,6 +502,8 @@ template <class RANDOM_ACCESS_ITERATOR>
     RANDOM_ACCESS_ITERATOR it_begin, RANDOM_ACCESS_ITERATOR it_end, unsigned shift_amnt
 )
 {
+    assert(it_begin < it_end);
+
     // Return early if no shift or no vector
     if (!shift_amnt) {
         return;
@@ -537,6 +547,9 @@ template <class RANDOM_ACCESS_ITERATOR>
     RANDOM_ACCESS_ITERATOR it_begin, RANDOM_ACCESS_ITERATOR it_end, unsigned shift_amnt
 )
 {
+    assert(it_begin < it_end);
+    assert(std::distance(it_begin, it_end) < shift_amnt / APY_LIMB_SIZE_BITS);
+
     // Return early if no shift or no vector
     if (!shift_amnt) {
         return;
@@ -576,6 +589,8 @@ static APY_INLINE void limb_vector_lsl_inner(
     unsigned int limb_shift
 )
 {
+    assert(it_begin < it_end);
+
     if (limb_skip) {
         for (auto it = it_end - 1; it != it_begin + limb_skip - 1; --it) {
             *it = *(it - limb_skip);
@@ -601,13 +616,15 @@ template <class RANDOM_ACCESS_ITERATOR>
     RANDOM_ACCESS_ITERATOR it_begin, RANDOM_ACCESS_ITERATOR it_end, unsigned shift_amnt
 )
 {
+    assert(it_begin < it_end);
+
     // Return early if no shift or no vector
     if (!shift_amnt) {
         return;
     }
 
-    unsigned limb_skip = shift_amnt / APY_LIMB_SIZE_BITS;
-    if (limb_skip >= std::distance(it_begin, it_end)) {
+    unsigned int limb_skip = shift_amnt / APY_LIMB_SIZE_BITS;
+    if (std::ptrdiff_t(limb_skip) >= std::distance(it_begin, it_end)) {
         std::fill(it_begin, it_end, 0);
         return; // early return
     }
@@ -622,6 +639,8 @@ template <class RANDOM_ACCESS_ITERATOR>
     RANDOM_ACCESS_ITERATOR it_begin, RANDOM_ACCESS_ITERATOR it_end, unsigned n
 )
 {
+    assert(it_begin <= it_end);
+
     unsigned bit_idx = n % APY_LIMB_SIZE_BITS;
     unsigned limb_idx = n / APY_LIMB_SIZE_BITS;
     std::size_t n_limbs = std::distance(it_begin, it_end);
@@ -678,6 +697,8 @@ template <class RANDOM_ACCESS_ITERATOR>
     RANDOM_ACCESS_ITERATOR it_begin, RANDOM_ACCESS_ITERATOR it_end, unsigned n
 )
 {
+    assert(it_begin < it_end);
+
     unsigned limb_idx = n / APY_LIMB_SIZE_BITS;
     std::size_t limbs = std::distance(it_begin, it_end);
     if (limb_idx < limbs) {
@@ -762,6 +783,7 @@ template <class RANDOM_ACCESS_ITERATOR>
 )
 {
     (void)cbegin_it;
+    assert(cbegin_it < cend_it);
     return apy_limb_signed_t(*std::prev(cend_it)) < 0;
 }
 
@@ -770,6 +792,7 @@ template <class RANDOM_ACCESS_ITERATOR>
 [[maybe_unused, nodiscard]] static APY_INLINE bool
 limb_vector_is_zero(RANDOM_ACCESS_ITERATOR cbegin_it, RANDOM_ACCESS_ITERATOR cend_it)
 {
+    assert(cbegin_it <= cend_it);
     return std::all_of(cbegin_it, cend_it, [](auto n) { return n == 0; });
 }
 
@@ -782,9 +805,11 @@ template <class RANDOM_ACCESS_ITERATOR>
 )
 {
     (void)cend_it;
-    const unsigned full_limbs = n / APY_LIMB_SIZE_BITS;
+    assert(cbegin_it < cend_it);
+    assert(std::distance(cbegin_it, cend_it) >= bits_to_limbs(n));
 
     // The full limbs can be reduced as full integers
+    const unsigned full_limbs = n / APY_LIMB_SIZE_BITS;
     for (auto limb_it = cbegin_it; limb_it != cbegin_it + full_limbs; ++limb_it) {
         if (*limb_it != 0) {
             return true;
@@ -812,6 +837,9 @@ template <class RANDOM_ACCESS_ITERATOR>
 )
 {
     (void)cend_it;
+    assert(cbegin_it < cend_it);
+    assert(std::distance(cbegin_it, cend_it) >= bits_to_limbs(n));
+
     unsigned bit_idx = n % APY_LIMB_SIZE_BITS;
     unsigned limb_idx = n / APY_LIMB_SIZE_BITS;
     apy_limb_t mask = apy_limb_t(1) << bit_idx;
@@ -827,6 +855,8 @@ template <class RANDOM_ACCESS_ITERATOR>
 )
 {
     (void)end_it;
+    assert(begin_it < end_it);
+
     unsigned bit_idx = n % APY_LIMB_SIZE_BITS;
     unsigned limb_idx = n / APY_LIMB_SIZE_BITS;
     apy_limb_t bit_mask = apy_limb_t(1) << bit_idx;
@@ -843,6 +873,7 @@ template <class RANDOM_ACCESS_ITERATOR_IN, class RANDOM_ACCESS_ITERATOR_OUT>
     RANDOM_ACCESS_ITERATOR_OUT res_it
 )
 {
+    assert(cbegin_it < cend_it);
     return apy_negate(cbegin_it, cend_it, res_it);
 }
 
@@ -852,6 +883,7 @@ template <class RANDOM_ACCESS_ITERATOR_IN>
     RANDOM_ACCESS_ITERATOR_IN cbegin_it, RANDOM_ACCESS_ITERATOR_IN cend_it
 )
 {
+    assert(cbegin_it < cend_it);
     return apy_inplace_negate(cbegin_it, cend_it);
 }
 
@@ -861,6 +893,7 @@ template <class RANDOM_ACCESS_ITERATOR_IN>
     RANDOM_ACCESS_ITERATOR_IN cbegin_it, RANDOM_ACCESS_ITERATOR_IN cend_it
 )
 {
+    assert(cbegin_it < cend_it);
     return apy_inplace_add_one_lsb(cbegin_it, cend_it);
 }
 
@@ -951,6 +984,8 @@ template <typename RANDOM_ACCESS_ITERATOR_IN, typename RANDOM_ACCESS_ITERATOR_OU
     RANDOM_ACCESS_ITERATOR_OUT dst_end
 )
 {
+    assert(src_begin <= src_end);
+    assert(dst_begin <= dst_end);
     std::size_t src_n = std::distance(src_begin, src_end);
     std::size_t dst_n = std::distance(dst_begin, dst_end);
     limb_vector_copy_n_sign_extend(src_begin, src_n, dst_begin, dst_n);
@@ -984,6 +1019,7 @@ template <typename RANDOM_ACCESS_IT>
 limb_vector_to_u64_vec(RANDOM_ACCESS_IT begin_it, RANDOM_ACCESS_IT end_it)
 {
     static_assert(APY_LIMB_SIZE_BITS == 64 || APY_LIMB_SIZE_BITS == 32);
+    assert(begin_it <= end_it);
 
     if constexpr (APY_LIMB_SIZE_BITS == 64) {
         return std::vector<uint64_t>(begin_it, end_it);
@@ -1008,6 +1044,8 @@ template <typename VEC_RETURN_TYPE, typename RANDOM_ACCESS_IT>
 limb_vector_from_u64_vec(RANDOM_ACCESS_IT begin_it, RANDOM_ACCESS_IT end_it)
 {
     static_assert(APY_LIMB_SIZE_BITS == 64 || APY_LIMB_SIZE_BITS == 32);
+    assert(begin_it <= end_it);
+
     if constexpr (APY_LIMB_SIZE_BITS == 64) {
         return VEC_RETURN_TYPE(begin_it, end_it);
     } else {
@@ -1096,6 +1134,7 @@ template <typename RANDOM_ACCESS_ITERATOR_INOUT>
     std::size_t itemsize
 )
 {
+    assert(begin_it <= end_it);
     auto n_items = std::distance(begin_it, end_it) / itemsize;
     for (std::size_t i = 0; i < (n_items + 1) / 2; i++) {
         auto it1 = begin_it + (i * itemsize);
@@ -1148,13 +1187,13 @@ CREATE_FUNCTOR_FROM_FUNC(apy_sub_n_functor, apy_subtraction_same_length);
 
 //! Mark nanobind Python-exposed parameter that does not support implicit conversions
 //! from other types.
-#define NB_NARG(name) nb::arg(name).noconvert()
+#define NB_NARG(...) nb::arg(__VA_ARGS__).noconvert()
 
 //! Mark a special double-underscore nanobind operator (e.g., `__add__`) the implements
 //! an arithmetic operation. In Python, when a bound function with this annotation is
 //! called with incompatible arguments, it will return `NotImplemented` rather than
 //! raising a `TypeError` as is default.
-#define NB_OP(args) nb::is_operator(args)
+#define NB_OP(...) nb::is_operator(__VA_ARGS__)
 
 //! Short-hand C++ arithmetic functors
 #define STD_ADD std::plus

--- a/src/python_util.h
+++ b/src/python_util.h
@@ -179,7 +179,7 @@ limb_vec_from_py_long_vec(const std::size_t count, const PyLongObject* py_long)
     );
 
     // Normalize
-    while (!limb_vec.empty() > 0 && (limb_vec.back() == 0)) {
+    while (!limb_vec.empty() && (limb_vec.back() == 0)) {
         limb_vec.pop_back();
     }
     return limb_vec;
@@ -276,7 +276,7 @@ template <class RANDOM_ACCESS_ITERATOR>
     }
 
     // Normalize
-    while (!limb_vec_abs.empty() > 0 && (limb_vec_abs.back() == 0)) {
+    while (!limb_vec_abs.empty() && (limb_vec_abs.back() == 0)) {
         limb_vec_abs.pop_back();
     }
     if (limb_vec_abs.size() == 0) {

